polaris-react/src/components/TopBar/components/UserMenu/UserMenu.tsx

```diff
@@ -29,6 +29,8 @@ export interface UserMenuProps {
   open: boolean;
   /** A callback function to handle opening and closing the user menu */
   onToggle(): void;
+  /** A custom activator that can be used when the default activator is not desired */
+  customActivator?: React.ReactNode;
```

Comment:
Agree ^. Also, there's a default activator already; passing a new one would be customizing it.

Reply:
@chloerice 

> Make this a layout change internal to the component instead of adding a customActivator prop, since the pattern is being updated across the board?

Making it an internal layout change is a good idea. Although I think we still need the `customActivator` prop because the release for the store switcher change is set for Editions; and also, Plus stores will not have their layout changed.

> Add an optional logo prop that renders a Thumbnail to the right of the name (instead of an avatar on the left) when provided, and make initials optional?

I think this depends on how much flexibility you'll want in Polaris. We can allow for both of them to be optional in a `prefix` / `suffix` type of way like we do in `ActionList`. On the other hand if we want to be stricter, we can just use one layout with the current `Avatar` component. The `avatar` prop already acts as a logo since it takes in a `url` that can be passed to the `Avatar` component.

Reply:
Thank you! FYI, I've made the new layout change internal and will temporarily pass in the old activator as `customActivator` until launch.